traffic_ops/traffic_ops_golang/servercapability/servercapability.go

```diff
@@ -100,6 +102,30 @@ func (v *TOServerCapability) Validate() error {
 	return util.JoinErrs(tovalidate.ToErrors(errs))
 }
 
-func (v *TOServerCapability) Read() ([]interface{}, error, error, int) { return api.GenericRead(v) }
-func (v *TOServerCapability) Create() (error, error, int)              { return api.GenericCreateNameBasedID(v) }
-func (v *TOServerCapability) Delete() (error, error, int)              { return api.GenericDelete(v) }
+func (v *TOServerCapability) Read(h map[string][]string) ([]interface{}, error, error, int) {
+	ims := h["If-Modified-Since"]
+	var modifiedSince time.Time
+	var res []interface{}
+
+	if ims == nil || len(ims) == 0 {
+		return api.GenericRead(v)
```

Comment:
There seems to be a lot of very similar code around reads. How difficult would it be to reduce the duplication by putting the "modified" logic inside `api.GenericRead` somehow?

Reply:
Yeah, I was just talking to Brennan about it, and thats the reason I marked this PR as a draft, because I'm trying to see if we can do it all in the GenericReader code. That way, we just make the call to the DB and dont have to copy the same code changes everywhere.